### PR TITLE
Add Windows Installation for Rancher, Update existing Installer

### PR DESCRIPTION
**Pull request by luthermonson**
#### Proposed Changes ####

We are adding support for installing the Windows Agent via Rancher with the rke2-installer.ps1. We have also resolved bugs related to the current install.ps1 in the root of the repo.

This will improve the experience for all users and allow better integration with Rancher provisioning.

#### Types of Changes ####
A new feature which is the RKE2-Installer.ps1 and bug fixes for installer.ps1.

#### Verification ####

Install via the installer.ps1 produces a Windows node that joins a cluster. The rancher installer will need have a build at the head of the repository, then create a custom cluster with Windows.

#### Linked Issues ####
https://github.com/rancher/rancher/issues/33625

### CODE REVIEW

File: install.ps1 [hunk 1]
```diff
@@ -509,7 +542,9 @@ function Install-AirgapTarball() {
 $STORAGE_URL = "https://storage.googleapis.com/rke2-ci-builds"
 $INSTALL_RKE2_GITHUB_URL = "https://github.com/rancher/rke2"
 $DEFAULT_TAR_PREFIX = "C:\usr\local"
+$INSTALL_RKE2_TAR_PREFIX = "C:\usr\local"
```
Comment (dereknola) [inline install.ps1:545]: Does this need to exist, as $DEFAULT_TAR_PREFIX is the same value above?

**Review by briandowns — APPROVED** [review]
LGTM overall but I'd like someone who knows powershell to give it a look.

File: install.ps1 (hunk 1 above)
Comment (luthermonson) [reply install.ps1:545]: its cause they both exist in the install.sh... dont disagree with you but im not sure it's my call

Comment (dereknola) [reply install.ps1:545]: Okay if they already are there, matching install.sh is fine.

**Review by dereknola — APPROVED** [review]

**Pull request merged**